crates/pallet-domains/src/staking.rs

```diff
@@ -621,10 +667,17 @@ pub(crate) fn do_switch_operator_domain<T: Config>(
         note_pending_staking_operation::<T>(operator.current_domain_id)?;
 
         ensure!(
-            operator.status == OperatorStatus::Registered,
+            *operator.status::<T>(operator_id) == OperatorStatus::Registered,
             Error::OperatorNotRegistered
         );
 
+        // Reject switching domain if there is unconfirmed ER submitted by this operator
+        // on the `current_domain_id`
+        ensure!(
+            !LatestSubmittedER::<T>::contains_key((operator.current_domain_id, operator_id)),
```

Comment:
Thinking on this further, there are two things we can do.
- Get operators for all unconfirmed ERs, either through Host function or through storage and then slash them immediately. We still prune the bad ers lazily though
- Or completely remove switching feature between domains

@jfrank-summit i'm leaning towards removing this switching the domains since with this change, its no different from operator deregistring, waiting and then registering to new domain unless we have a hard requirement to have it though, then we need to do what I suggested in the first point of slashing all operators immediately so that Lazy slashing wont be necessary

Reply:
>Can you explain the reason why we do not want to allow switching the operators?

We are using the `LatestSubmittedER ` to check if an operator is pending to slash, `LatestSubmittedER` is indexed by `(domain_id, operator_id)` we can only check if the operator have submitted bad ER in the current domain but can't know if the operator in the previous domain.

I planned to handle the switching domain in a separate PR after this one is merged with what Daria suggested in https://github.com/subspace/subspace/pull/2560#discussion_r1503839500, but if we plan to remove switching domain then we are all good and it will be great.

>then we need to do what I suggested in the first point of slashing all operators immediately so that Lazy slashing wont be necessary

To slash operator immediately we need to iterate at most 14_400 number of ERs, duplicate the operator, and then slash all of them. It will still cause a lot of weight.

Reply:
I am ok with removing domain switching for operators.